crates/rattler_networking/src/s3_middleware.rs

```diff
@@ -80,41 +83,62 @@ impl S3 {
                         secret_access_key,
                         session_token,
                     }),
-                ) => aws_sdk_s3::config::Builder::from(&sdk_config)
-                    .endpoint_url(endpoint_url)
-                    .region(aws_sdk_s3::config::Region::new(region))
-                    .force_path_style(force_path_style)
-                    .credentials_provider(aws_sdk_s3::config::Credentials::new(
-                        access_key_id,
-                        secret_access_key,
-                        session_token,
-                        None,
-                        "pixi",
-                    )),
+                ) => {
+                    let sdk_config = aws_config::defaults(BehaviorVersion::latest()).load().await;
+                    aws_sdk_s3::config::Builder::from(&sdk_config)
+                        .endpoint_url(endpoint_url)
+                        .region(aws_sdk_s3::config::Region::new(region))
+                        .force_path_style(force_path_style)
+                        .credentials_provider(aws_sdk_s3::config::Credentials::new(
+                            access_key_id,
+                            secret_access_key,
+                            session_token,
+                            None,
+                            "pixi",
+                        ))
+                }
                 (_, Some(_)) => {
-                    panic!("Unsupported authentication method"); // todo proper error message
+                    return Err(anyhow::anyhow!("Unsupported authentication method"));
+                }
+                (_, None) => {
+                    eprintln!("No authentication found");
```

Comment:
this should probably be a logging message instead of stderr

Reply:
Done